### PR TITLE
remove properties from RayToEntityIntersectionResult

### PR DESCRIPTION
**Pull request by ZappoMan**
This obsoletes the `properties` member of the result from `Entities.findRayIntersection`
We are removing it because it's not usually used, and it's to expensive to produce when it's not actually used. For scripts that need access to this information they can call `Entities.getEntityProperties(result.entityID)`

### CODE REVIEW

**Comment by thoys** [comment]
Hmm this might break some marketplace scripts. Let me check

**Comment by ZappoMan** [comment]
@thoys - let's find out, and fix any marketplace scripts doing this...

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9e71801ae170cf2359304b8c8d8d612bbe854128 available at: https://gist.github.com/9739b90fba04d96f6a9dd67c827ac9ca<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9796-9e71801.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9796-9e71801.dmg

**Comment by Nex-Pro** [comment]
testing this

**Comment by Nex-Pro** [comment]
### IMO FAIL

I compared this to 6102.

In this PR I had to use createPistol.js to get a pistol then pistol.js to use it.
The pistol sound was bad when the shots were hitting close by. ( choppy ).
It was easy to get the pistol into full auto mode which would eventually cause me to disconnect for a moment ( black screen ) and reconnect. Then the gun would de-render.
No such problems in 6102. ( I only needed createPistol.js )


Also noticed in both builds that if scaling of a grabbed entity is initiated then the right trigger is pulled the avatar also scales. So both Ent and Av are scaling simultaneously. Is this a bug or intended ?

In both builds the gun equips the left hand with the pinky at the hammer.

File: script-archive/example/games/hydraGrabHockey.js [hunk 1]
```diff
@@ -190,7 +190,10 @@ function controller(side) {
       direction: Vec3.normalize(Vec3.subtract(this.tipPosition, this.palmPosition))
     };
     var intersection = getRayIntersection(pickRay, true);
-    if (intersection.intersects && intersection.properties.dynamic) {
+
+    var isDynamic = Entites.getEntityProperties(intersection.entityID,"dynamic").dynamic;
```
Comment (hyperlogic) [inline script-archive/example/games/hydraGrabHockey.js]: should be a space between the comma and "dynamic".

File: script-archive/example/games/grabHockey.js [hunk 2]
```diff
@@ -142,7 +142,8 @@ function mousePressEvent(event) {
   if (!pickResults.intersects) {
     return;
   }
-  if (pickResults.properties.dynamic) {
+  var isDynamic = Entites.getEntityProperties(pickResults.entityID,"dynamic").dynamic;
```
Comment (hyperlogic) [inline script-archive/example/games/grabHockey.js]: Should be a space between the comma and "dynamic"

File: script-archive/painting/whiteboard/whiteboardEntityScript.js [hunk 3]
```diff
@@ -77,6 +77,8 @@
             //Comment out above line and uncomment below line to see difference in performance between using a whitelist, and not using one
             // this.intersection = Entities.findRayIntersection(pickRay, true);
 
+            var type = Entites.getEntityProperties(this.intersection.entityID,"type").type;
```
Comment (hyperlogic) [inline script-archive/painting/whiteboard/whiteboardEntityScript.js]: Space between comma and "type"

File: script-archive/vrShop/item/shopItemGrab.js [hunk 4]
```diff
@@ -382,7 +382,10 @@ function MyController(hand) {
 
                 var grabbableData = getEntityCustomData(GRABBABLE_DATA_KEY, intersection.entityID, DEFAULT_GRABBABLE_DATA);
 
-                if (intersection.properties.name == "Grab Debug Entity") {
+                var properties = Entites.getEntityProperties(intersection.entityID,["locked","name"]);
```
Comment (hyperlogic) [inline script-archive/vrShop/item/shopItemGrab.js]: Space between comma and "name".  Also space after comma and ["locked"

File: scripts/system/controllers/grab.js [hunk 5]
```diff
@@ -343,7 +343,8 @@ Grabber.prototype.pressEvent = function(event) {
         return;
     }
 
-    if (!pickResults.properties.dynamic) {
+    var isDynamic = Entites.getEntityProperties(pickResults.entityID,"dynamic").dynamic;
```
Comment (hyperlogic) [inline scripts/system/controllers/grab.js]: space between comma and "dynamic"

File: scripts/tutorials/entity_scripts/pistol.js [hunk 6]
```diff
@@ -151,8 +151,10 @@
                         });
                     }, randFloat(10, 200));
                 }
-                if (intersection.properties.dynamic === 1) {
-                    // Any dynaic entity can be shot
+                var isDynamic = Entites.getEntityProperties(intersection.entityID,"dynamic").dynamic;
```
Comment (hyperlogic) [inline scripts/tutorials/entity_scripts/pistol.js]: space between comma and "dynamic"

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9e287d4b59ec93a24383e84406067a104870b584 available at: https://gist.github.com/9739b90fba04d96f6a9dd67c827ac9ca<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9796-9e287d4.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9796-9e287d4.dmg

**Comment by ZappoMan** [comment]
test this please

**Comment by ZappoMan** [comment]
@Nex-Pro - hmm... your fail comments don't really make sense to me, I didn't change anything about the system that would result in those behaviors... it's really only ray picking features that would be changed and potentially break.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9e287d4b59ec93a24383e84406067a104870b584 available at: https://gist.github.com/9739b90fba04d96f6a9dd67c827ac9ca<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9796-9e287d4.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9796-9e287d4.dmg

**Comment by Nex-Pro** [comment]
@ZappoMan you asked if pistol.js still works.  It doesn't work as it should. Maybe something else broke it. I just wrote what I observed. This was easily repeatable. I could not get it to happen in the other build.

**Comment by ZappoMan** [comment]
Test Plan:

To Test:
- Smoke test
- make sure the following scripts still work:
    - grab.js
- using the console run the following script... verify that when you aim at a dynamic object and shoot the gun, that the object will be "hit" and bounce away as if hit by the bullet. (any other odd behavior in the gun, is not a fail, as this PR does not side effect any other behavior).
```
var center = Vec3.sum(MyAvatar.position, Vec3.multiply(1.5, Quat.getFront(Camera.getOrientation())));
var SCRIPT_URL = "https://raw.githubusercontent.com/ZappoMan/hifi/9e287d4b59ec93a24383e84406067a104870b584/scripts/tutorials/entity_scripts/pistol.js";
var MODEL_URL = "http://hifi-production.s3.amazonaws.com/tutorials/pistol/gun.fbx";
var COLLISION_SOUND_URL = 'http://hifi-production.s3.amazonaws.com/tutorials/pistol/drop.wav'

var pistolProperties = {
  type: 'Model',
  modelURL: MODEL_URL,
  position: center,
  dimensions: {
    x: 0.05,
    y: 0.23,
    z: 0.36
  },
  script: SCRIPT_URL,
  color: {
    red: 200,
    green: 0,
    blue: 20
  },
  shapeType: 'box',
  dynamic: true,
  gravity: {
    x: 0,
    y: -5.0,
    z: 0
  },
  lifetime: 3600,
  restitution: 0,
  damping: 0.5,
  collisionSoundURL: COLLISION_SOUND_URL,
  userData: JSON.stringify({
    grabbableKey: {
      invertSolidWhileHeld: true
    },
    wearable: {
      joints: {
        RightHand: [{
          x: 0.07079616189002991,
          y: 0.20177987217903137,
          z: 0.06374628841876984
        }, {
          x: -0.5863648653030396,
          y: -0.46007341146469116,
          z: 0.46949487924575806,
          w: -0.4733745753765106
        }],
        LeftHand: [{
          x: 0.1802254319190979,
          y: 0.13442856073379517,
          z: 0.08504903316497803
        }, {
          x: 0.2198076844215393,
          y: -0.7377811074256897,
          z: 0.2780133783817291,
          w: 0.574519157409668
        }]
      }
    }
  })
};

var pistol = Entities.addEntity(pistolProperties);
```

**Comment by Nex-Pro** [comment]
@ZappoMan this does not work. I pasted that code into the console in this build. I get a pistol.
I then got the ball collection from the market as targets.
With the gun equipped I fire at the balls.

taking into account your note:
 (any other odd behavior in the gun, is not a fail, as this PR does not side effect any other behavior).

Results:

The gun gets locked in some strange auto-fire mode

The balls do not move when hit.

The balls are dynamic and react to contact with each other and other objects.

The balls move in reaction to being hit with the ping-pong ball gun ( albeit very slightly )

I did not proceed to smoke test this and I await your comments.

See attached gifs.

[hifi-snap-by-Piper-Peppercorn-on-2017-03-22_13-47-19.zip](https://github.com/highfidelity/hifi/files/861600/hifi-snap-by-Piper-Peppercorn-on-2017-03-22_13-47-19.zip)
[hifi-snap-by-Piper-Peppercorn-on-2017-03-22_13-48-03.zip](https://github.com/highfidelity/hifi/files/861605/hifi-snap-by-Piper-Peppercorn-on-2017-03-22_13-48-03.zip)

**Comment by hifi-gustavo** [comment]
Build console output for commit: 246f377a070d87514208b3f2bf514f962ee74a09 available at: https://gist.github.com/9739b90fba04d96f6a9dd67c827ac9ca<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9796-246f377.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9796-246f377.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: bc331af6a0483459b1f21637a299581ce52b54e4 available at: https://gist.github.com/9739b90fba04d96f6a9dd67c827ac9ca<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9796-bc331af.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9796-bc331af.dmg

**Comment by ZappoMan** [comment]
@Nex-Pro - ok... thanks for being persistent... there was indeed a bug... I am pretty sure it works now.

Note: on the testing of `grab.js`... this script is the 2D screen mouse grabbing... it's normally part of default scripts. so you don't need to run it, manually. It is what allows you to move/grab things with your mouse while not in edit mode. Grab only allows you to grab/move dynamic things... the test is to verify that it still works.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 52335e20a2d7331fc901ff9b40178abbe58271fd available at: https://gist.github.com/9739b90fba04d96f6a9dd67c827ac9ca<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9796-52335e2.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9796-52335e2.dmg

**Comment by Nex-Pro** [comment]
Giving this a prod.

**Comment by Nex-Pro** [comment]
#9796

## IMO PASS

but please read  observations.

Testing platform: Intel workstation.
OS: Win10 pro 64
CPU: i7 4.0 ghz
RAM: 16gb
GPU: Nvidia GTX1070 8gb VRam
HMD: Htc Vive.


## QA Test
**QA Tester**: Piper.Peppercorn

**localhost 'createTestBlocks.js' Result**: SUCCESS
**localhost ATP Upload + Edit Result**: SUCCESS
**Echo Server Audio**: SUCCESS
**localhost 'harmonicOscillator.js' Result**: FAILURE
**localhost 'reverbTest.js' Result**: SUCCESS
**localhost 'playaPerformanceTest.js' Result**: SUCCESS
**dev-playa 'playaPerformanceTest.js' Result**: SUCCESS
**dev-demo 'playaPerformanceTest.js' Result**: N/A
**QA Pass**: SUCCESS

**Additional Test Results + Observations**:

Unsure if relevant but was unable to complete HMD tutorial
Was not possible to ignite fuse. Have never had this issue before. I am aware that there is an invisible box around the ball that represents the end of the fuse. It was present, I checked in edit mode.
Problem persisted across restarts in this build.


**Domain** |localhost |dev-playa |
---|---|---|---
**Render Rate**: |89 |89 |
**Present Rate**: |90 |89 |
**Present New Rate**: |89 |100 |
**CPU Buffer Count**: |2151 |3592 |
**GPU Buffer Count**: |880 |911 |
**CPU Memory Usage**: |22341648 |61477424 |
**GPU Memory Usage**: |16545072 |39379696 |
**CPU Texture Count**: |230 |226 |
**GPU Texture Count**: |201 |164 |
**GPU Transfer Count**: |0 |0 |
**CPU Texture Memory Usage**: |0 |0 |
**GPU Texture Memory Usage**: |665396207 |685113045 |
**GPU Virtual Memory Usage**: |0|0|
**Triangle Count**: |90833 |1319298 |
**Triangle Rate**: |10092555 |131929800 |
**API Drawcall Count**: |22 |128 |
**Drawcall Count**: |31 |242 |
**Drawcall Rate**: |3444 |24200 |
**State Changes Texture Count**:|47 |153 |
**Pipeline Count**: |27 |32 |
**Item Opaques**: |12 |186 |
**Item Translucents**: |3 |11 |
**Item Lights**: |0 |30 |
**Timing Opaques**: |0.068 |0.559 |
**Timing Translucents**: |0.007 |0.035 |
**Timing Lighting**: |0.009 |0.019 |
**Render Frame**: |0.415 |1.841 |

**Comment by Nex-Pro** [comment]
addendum: When I went into edit mode by activating the hud using the menu I could not interact with any dialogs using lasers. ( in failed tutorial ).

**Pull request merged**